src/libertem/udf/holography.py

```diff
@@ -0,0 +1,199 @@
+
+        f_freq = freq_array(out_shape)
+        aperture = aperture_function(f_freq, sb_size, sb_smoothness)
+
+        y_min = int(sy / 2 - oy / 2)
+        y_max = int(sy / 2 + oy / 2)
+        x_min = int(sx / 2 - ox / 2)
+        x_max = int(sx / 2 + oy / 2)
+        slice_fft = ([slice(y_min, y_max), slice(x_min, x_max)])
+
+        kwargs = {
+            'aperture': aperture,
+            'slice': slice_fft
+        }
+        return kwargs
+
+    def process_frame(self, frame):
+        """
+        Reconstructs holograms outputting results into 'wave'
+
+        Parameters
+        ----------
+        frame
+           single frame (hologram) of the data
+        """
+        if not self.params.precision:
+            frame = frame.astype(np.float32)
+        # size_x, size_y = self.params.out_shape
+        frame_size = self.meta.partition_shape.sig
+        sb_pos = self.params.sb_position
+        aperture = self.task_data.aperture
+        slice_fft = self.task_data.slice
+
+        fft_frame = fft2(frame) / np.prod(frame_size)
+        fft_frame = np.roll(fft_frame, sb_pos[0], axis=0)
+        fft_frame = np.roll(fft_frame, sb_pos[1], axis=1)
+
+        fft_frame = fftshift(fftshift(fft_frame)[slice_fft])
+
+        fft_frame = fft_frame * aperture
+
+        wav = ifft2(fft_frame) * np.prod(frame_size)
+        self.results.wave[:] = wav
+
+
+def reconstruct_holograms(ctx, dataset,
+                          out_shape,
+                          sb_position,
+                          sb_size,
+                          sb_smoothness=0.05,
+                          precision=True):
+    """
+    Reconstruct hologram using standard Fourier method
+
+    Parameters
+    ----------
+    ctx
+        Context class that contains methods for loading datasets, creating jobs on them
+        and running them
+
+    dataset
+        dataset to work on
+
+    out_shape : (int, int)
+        Shape of the outputted complex wave image
+
+    sb_position : tuple, or vector
+        Coordinates of sideband position with respect to non-shifted FFT of a hologram
+
+    sb_size : float
+        Radius of side band filter in pixels
+
+    sb_smoothness : float, optional (Default: 0.05)
+        Fraction of `sb_size` over which the edge of the filter aperture to be smoothed
+
+    precision : bool, optional, (Default: True)
+        Defines precision of the reconstruction, True for complex128 for the resulting
+        complex wave, otherwise results will be complex64
+    Returns
+    -------
+    wave : ndarray
+        Reconstructed complex wave function
+    """
+
+    holo_job = HoloReconstructUDF(out_shape=out_shape,
+                                  sb_position=sb_position,
+                                  sb_size=sb_size,
+                                  sb_smoothness=sb_smoothness,
+                                  precision=precision)
+    res = ctx.run_udf(dataset=dataset, udf=holo_job)
+
+    return res['wave'].data
```

Comment:
@sk1p With regards to the doc strings, at the moment I'm intended to put the description of pseudo-return and the example into global docstring, would it be fine

Reply:
Should be fine. If possible, the example in the docstring should be runnable, so that a run of `pytest --doctest-modules --ignore=src/libertem/win_tweaks.py src/libertem/` succeeds. in the doctests, you have a few objects always available, such as a dataset and a `Context` (see [the toplevel `conftest.py` file](https://github.com/LiberTEM/LiberTEM/blob/master/conftest.py#L163), where you can also add your own)